modules/src/events.rs

```diff
@@ -0,0 +1,248 @@
+use crate::ics02_client::events as ClientEvents;
+use crate::ics02_client::events::NewBlock;
+use crate::ics03_connection::events as ConnectionEvents;
+use crate::ics04_channel::events as ChannelEvents;
+use crate::ics20_fungible_token_transfer::events as TransferEvents;
+use serde_derive::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use tendermint::block;
+use tendermint::rpc::event_listener::{ResultEvent, TMEventData};
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum IBCEvent {
+    NewBlock(NewBlock),
+
+    CreateClient(ClientEvents::CreateClient),
+    UpdateClient(ClientEvents::UpdateClient),
+    ClientMisbehavior(ClientEvents::ClientMisbehavior),
+
+    OpenInitConnection(ConnectionEvents::OpenInit),
+    OpenTryConnection(ConnectionEvents::OpenTry),
+    OpenAckConnection(ConnectionEvents::OpenAck),
+    OpenConfirmConnection(ConnectionEvents::OpenConfirm),
+
+    OpenInitChannel(ChannelEvents::OpenInit),
+    OpenTryChannel(ChannelEvents::OpenTry),
+    OpenAckChannel(ChannelEvents::OpenAck),
+    OpenConfirmChannel(ChannelEvents::OpenConfirm),
+    CloseInitChannel(ChannelEvents::CloseInit),
+    CloseConfirmChannel(ChannelEvents::CloseConfirm),
+
+    SendPacketChannel(ChannelEvents::SendPacket),
+    ReceivePacketChannel(ChannelEvents::ReceivePacket),
+    AcknowledgePacketChannel(ChannelEvents::AcknowledgePacket),
+    CleanupPacketChannel(ChannelEvents::CleanupPacket),
+    TimeoutPacketChannel(ChannelEvents::TimeoutPacket),
+
+    TimeoutTransfer(TransferEvents::Timeout),
+    PacketTransfer(TransferEvents::Packet),
+    ChannelClosedTransfer(TransferEvents::ChannelClosed),
+}
+
+impl IBCEvent {
+    pub fn to_json(&self) -> String {
+        serde_json::to_string(self).unwrap()
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct TryObject {
```

Comment:
Perhaps `RawObject` is a better name here, for consistency with this pattern in `tendermint-rs`?

Reply:
changed